test/t/test_scp.py

```diff
@@ -95,3 +96,32 @@ def test_remote_path_with_nullglob(self, completion):
     )
     def test_remote_path_with_failglob(self, completion):
         assert not completion
+
+    @pytest.fixture(scope="class")
+    def ssh_mock(self, request, bash):
+        with bash_env_saved(bash) as bash_env:
+            bash_env.write_variable(
+                "PATH",
+                "$PWD/scp/bin:$PATH",
+                quote=False,
+            )
+            yield
```

Comment:
After another look, I feel this doesn't need to be a pytest fixture. This fixture is used by only one test, so this temporary value of `PATH` can be just set inside `test_xfunc_remote_files`.

Reply:
I was actually wondering whether the lifetime of the generator fixture `ssh_mock` (and thus the `with` statement inside the generator) is well-defined and matches our expectations. It now seems to work, but this can be broken when we add multiple tests relying on the same fixture or when another condition is changed. Can you explain why this works, in particular, how the ordering of our `get_env()` check and `bash_env_saved.__exit__` is stably sequenced?

Reply:
>After another look, I feel this doesn't need to be a pytest fixture. This fixture is used by only one test, so this temporary value of PATH can be just set inside test_xfunc_remote_files.

I get that, I was thinking of adding a a test for the new `-l` (less escaping) option here, which will need basically the same steps. Maybe I can just do both in the same test instead?

> I was actually wondering whether the lifetime of the generator fixture `ssh_mock` (and thus the `with` statement inside the generator) is well-defined and matches our expectations. It now seems to work, but this can be broken when we add multiple tests relying on the same fixture or when another condition is changed. Can you explain why this works, in particular, how the ordering of our `get_env()` check and `bash_env_saved.__exit__` is stably sequenced?

Everything that happens after the yield (the finalization) happens in the [reverse order of calling the fixtures](https://docs.pytest.org/en/6.2.x/fixture.html#yield-fixtures-recommended). Since `ssh_mock` depends on the `bash` fixture, `bash` gets called first when running, and last in the finalization, so `get_env(0` will happen after `bash_env_saved.__exit__` is called.

Reply:
> Maybe I can just do both in the same test instead?

I think that is better because other tests in the same class may also be affected by this temporary variable change.

> Everything that happens after the yield (the finalization) happens in the [reverse order of calling the fixtures](https://docs.pytest.org/en/6.2.x/fixture.html#yield-fixtures-recommended).

Thank you! It's good to know that!

> Since `ssh_mock` depends on the `bash` fixture,

I also missed this point...

Reply:
OK, for now I removed the `ssh_mock`, if we will need it we can add it later